File: 백준_자바스크립트/1912.js
const io = process.platform === "linux" ? "/dev/stdin" : "_input.txt";
const fs = require("fs");

let [N, arr] = fs.readFileSync(io).toString().trim().split("\n");
arr = arr.split(" ").map(Number);

let max_sum = -Infinity;
let current_max_sum = -Infinity;

for (let i = 0; i < N; i++) {
    current_max_sum = current_max_sum + arr[i] > arr[i] ? current_max_sum + arr[i] : arr[i];
    max_sum = current_max_sum > max_sum ? current_max_sum : max_sum;
}

console.log(max_sum);

File: 백준/백준_파이썬/1912.py
N = int(input())

arr = list(map(int,input().split()))

max_sum = float("-inf") 
cur_sum = float("-inf") # 이때까지 최대 합

for i in range(N):
    cur_sum = max(arr[i], cur_sum + arr[i]) # 이때까지 최대합 + 요소 추가 vs 이때까지 최대합
    max_sum = max(max_sum, cur_sum)

print(max_sum)